runtime/src/mm/cpp/Memory.cpp

```diff
@@ -366,7 +375,22 @@ extern "C" void AdoptReferenceFromSharedVariable(ObjHeader* object) {
     // Nothing to do.
 }
 
-void CheckGlobalsAccessible() {
+extern "C" void CheckGlobalsAccessible() {
     // TODO: Remove when legacy MM is gone.
     // Always accessible
 }
+
+extern "C" RUNTIME_NOTHROW void Kotlin_mm_safePointFunctionEpilogue() {
+    auto* threadData = mm::ThreadRegistry::Instance().CurrentThreadData();
```

Comment:
I guess eventually it would make a lot of sense (for performance) to take `threadData` as a parameter.
For example, almost each non-trivial function would have at least a couple of runtime calls requiring thread data -- `EnterFrame`, `LeaveFrame`, `Kotlin_mm_safePointFunctionEpilogue`.

Reply:
I think so too. However, it might turn out that we'll have to split `ThreadData` into parts and pass them separately (or acquire them via TLS) to avoid an extra indirection from dereferencing `ThreadData`.